src/test/ui/structs-enums/type-sizes.rs

```diff
@@ -120,6 +120,16 @@ pub enum AlwaysTaggedBecauseItHasNoNiche {
     B
 }
 
+pub enum NicheFilledMultipleFields {
+    A(bool, u8),
+    B(u8),
+    C(u8),
+    D(bool),
+    E,
+    F,
+    G,
+}
+
 pub fn main() {
```

Comment:
#70477 has a few additional tests that would be nice to see here: https://github.com/rust-lang/rust/pull/70477/files#diff-fa3834a66481b6a1a59841622cd3ae9f2a9f87b994aa33c3fbf30e1e2dbecda3R187-R212